lib/SILOptimizer/Mandatory/TFDeabstraction.cpp

```diff
@@ -1905,8 +1905,31 @@ void TFDeabstraction::formGraphOp(SILTensorOpInfo &opInfo,
                           "' is not allowed");
           return;
         }
+
+        // Check to see if it was constant foldable.  If so, we can turn this
+        // into a Const node to avoid a send.
+        auto it = constants.find(operand);
+        if (it != constants.end()) {
+          // Dig the element type out of the TensorHandle result type.
+          auto eltType =
+            getTensorHandleElementType(inst->getType().getASTType());
+          auto int32Ty =
```

Comment:
add a comment that this is for shape?

or, given `SymbolicValue::getArray({}, int32Ty, allocator)` is used in a few places, consider making it a util method. the method `addScalarShapeArrayAttr()` in TFPartition.cpp uses such code as well.

Reply:
Done.